python/ray/_private/resource_spec.py

```diff
@@ -183,6 +189,21 @@ def resolve(self, is_head, node_ip_address=None):
                     "when calling ray.init() or ray start.")
                 object_store_memory = max_cap
 
+        if sys.platform == "darwin" and object_store_memory > \
+                best_perf_capacity:
+            logger.warning(
```

Comment:
Could we make this an error, guarded by RAY_OBJECT_STORE_ALLOW_BAD_OSX_CONFIG?

I.e., mirroring this other error:
```
                raise ValueError(
                    "The configured object store size ({} GB) exceeds "
                    "/dev/shm size ({} GB). This will harm performance. "
                    "Consider deleting files in /dev/shm or increasing its "
                    "size with "
                    "--shm-size in Docker. To ignore this warning, "
                    "set RAY_OBJECT_STORE_ALLOW_SLOW_STORAGE=1.".format(
                        object_store_memory / 1e9, shm_avail / 1e9))
```